sdl_android/src/main/java/com/smartdevicelink/api/audio/AudioStreamManager.java

```diff
@@ -0,0 +1,192 @@
+package com.smartdevicelink.api.audio;
+
+import android.net.rtp.AudioStream;
+import android.os.Build;
+import android.support.annotation.IntDef;
+import android.support.annotation.NonNull;
+import android.support.annotation.RequiresApi;
+import android.support.annotation.StringDef;
+import android.util.Log;
+
+import com.smartdevicelink.SdlConnection.SdlSession;
+import com.smartdevicelink.protocol.enums.SessionType;
+import com.smartdevicelink.proxy.interfaces.IAudioStreamListener;
+import com.smartdevicelink.proxy.interfaces.ISdl;
+import com.smartdevicelink.proxy.interfaces.ISdlServiceListener;
+import com.smartdevicelink.proxy.rpc.enums.BitsPerSample;
+import com.smartdevicelink.proxy.rpc.enums.SamplingRate;
+
+import java.io.File;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.util.LinkedList;
+import java.util.Queue;
+
+@RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
+public class AudioStreamManager implements ISdlServiceListener {
+    private static final String TAG = AudioStreamManager.class.getSimpleName();
+
+    private ISdl sdlInterface;
+    private IAudioStreamListener sdlAudioStream;
+    private int sdlSampleRate;
+    private @SampleType int sdlSampleType;
+    private final Queue<BaseAudioDecoder> queue;
+    private boolean didRequestShutdown = false;
+
+    public AudioStreamManager(@NonNull ISdl sdlInterface, @NonNull @SamplingRate String sampleRate, @NonNull @BitsPerSample int sampleType) {
+        this.sdlInterface = sdlInterface;
+        this.queue = new LinkedList<>();
+
+        switch (sampleRate) {
+            case SamplingRate.EIGHT_KHZ:
+                sdlSampleRate = 8000;
+                break;
+            case SamplingRate.SIXTEEN_KHZ:
+                sdlSampleRate = 16000;
+                break;
+            case SamplingRate.TWENTY_TWO_KHZ:
+                // common sample rate is 22050, not 22000
+                // see https://en.wikipedia.org/wiki/Sampling_(signal_processing)#Audio_sampling
+                sdlSampleRate = 22050;
+                break;
+            case SamplingRate.FOURTY_FOUR_KHX:
+                // 2x 22050 is 44100
+                // see https://en.wikipedia.org/wiki/Sampling_(signal_processing)#Audio_sampling
+                sdlSampleRate = 44100;
+                break;
+        }
+
+        switch (sampleType) {
+            case BitsPerSample.EIGHT_BIT:
+                sdlSampleType = SampleType.UNSIGNED_8_BIT;
+                break;
+            case BitsPerSample.SIXTEEN_BIT:
+                sdlSampleType = SampleType.SIGNED_16_BIT;
+                break;
+        }
+    }
+
+    public void startAudioService(boolean encrypted) {
+        if (sdlInterface != null && sdlInterface.isConnected()) {
+            sdlInterface.addServiceListener(SessionType.PCM, this);
+            sdlInterface.startAudioService(encrypted);
+        }
+    }
+
+    public void stopAudioService() {

```

Comment:
Change name to stop() based on proposal https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0113-audiostreammanager.md#android

Reply:
I don't think this is a good idea because "stop" is too confusing. Does it start the audio service or the stream? the decoder? I decided to give it a more specific name (just like the video manager is doing).

Reply:
Add javadoc to clarify